src/main/java/frc/robot/commands/IntakeToAmperCommand.java

```diff
@@ -11,16 +39,82 @@ public class IntakeToAmperCommand extends Command {
+
+private final ChassisSpeedsRateLimiter rateLimiter = new ChassisSpeedsRateLimiter(
+      TRANSLATION_RATE_LIMIT.in(MetersPerSecondPerSecond), ROTATION_RATE_LIMIT.in(RadiansPerSecond.per(Second)));
+
+  private final SwerveRequest.FieldCentric drive = new SwerveRequest.FieldCentric()
+      .withDeadband(MAX_VELOCITY.times(DEADBAND).in(MetersPerSecond))
+      .withRotationalDeadband(MAX_ANGULAR_VELOCITY.times(DEADBAND).in(RadiansPerSecond))
+      .withDriveRequestType(DriveRequestType.Velocity)
+      .withSteerRequestType(SteerRequestType.MotionMagic);
+
+  // Desired chassis speeds. Defined here to prevent reallocation.
+  private ChassisSpeeds desiredChassisSpeeds = new ChassisSpeeds();
 
-  public IntakeToAmperCommand(IntakeSubsystem intakeSubsystem, AmperSubsystem amperSubsystem) {
+
+  public IntakeToAmperCommand(
+  IntakeSubsystem intakeSubsystem,  
+  AmperSubsystem amperSubsystem,
+  CommandSwerveDrivetrain drivetrainSubsystem,
+  LimelightSubsystem limelightSubsystem, 
+  Supplier<Measure<Velocity<Distance>>> translationXSupplier,
+  Supplier<Measure<Velocity<Distance>>> translationYSupplier,
+  Supplier<Measure<Velocity<Angle>>> rotationSupplier,
+  LimelightResults limelightResults,
+  LimelightCalcs limelightCalcs,
+  LimelightDetectorTarget limelightDetectorTarget) {
     this.intakeSubsystem = intakeSubsystem;
     this.amperSubsystem = amperSubsystem;
+    this.drivetrainSubsystem = drivetrainSubsystem;
+    this.translationXSupplier = translationXSupplier;
+    this.translationYSupplier = translationYSupplier;
+    this.rotationSupplier = rotationSupplier;
+    this.limelightResults = limelightResults;
+    this.limelightCalcs = limelightCalcs;
+    this.limelightDetectorTarget = limelightDetectorTarget;
+
+    addRequirements(intakeSubsystem, amperSubsystem, drivetrainSubsystem);
+  }
+
+public Rotation2d getTargetRotation() {
+  return limelightCalcs.getRobotRelativeTargetInfo(
+  limelightDetectorTarget.targetXDegrees, limelightDetectorTarget.targetYDegrees).angle;
+}
 
-    addRequirements(intakeSubsystem, amperSubsystem);
+  @Override 
+  public void initialize() {
+    // Reset the slew rate limiters, in case the robot is already moving
+    rateLimiter.reset(drivetrainSubsystem.getCurrentFieldChassisSpeeds());
   }
 
   @Override
   public void execute() {
+    desiredChassisSpeeds.vxMetersPerSecond = translationXSupplier.get().in(MetersPerSecond);
+    desiredChassisSpeeds.vyMetersPerSecond = translationYSupplier.get().in(MetersPerSecond);
+    desiredChassisSpeeds.omegaRadiansPerSecond = rotationSupplier.get().in(RadiansPerSecond);
+    var limitedCassisSpeeds = rateLimiter.calculate(desiredChassisSpeeds);
+    drivetrainSubsystem.setControl(drive
+        .withVelocityX(limitedCassisSpeeds.vxMetersPerSecond)
+        .withVelocityY(limitedCassisSpeeds.vyMetersPerSecond)
+        .withRotationalRate(limitedCassisSpeeds.omegaRadiansPerSecond));
+    if (limelightResults.valid == true) {
```

Comment:
This is OK as is, but I have two tips:
1) There is no need to use `== true`, `valid` is already a boolean, so you can just do `if (limelightResults.valid)`
2) It's a best practice in cases like this (if you were to keep the `== true`) to put the literal on the left. It's a bit quirky, but it will help you catch a mistake at compile time instead of runtime. Consider this mistakes:
```java
if (limelightResults.valid = true) { // always true
  System.out.println("true");
}
```
This code will always print `true`. The bug is that the _assignment_ `=` was used instead of _equality check_ `==`. This will assign `limelightresults.valid` equal to `true` and return the value `true`, so the condition is also always `true`.

If we flip the same mistake around, we'll get a compile error because you cannot assign a value to the literal `true`:
```java
if (true = limelightResults.valid ) {  // compile error
  System.out.println("true");
}
```
 
Of these two options, I prefer number 1